dbt/include/global_project/macros/core.sql

```diff
@@ -7,12 +10,13 @@
       {{ write(sql) }}
     {%- endif -%}
 
-    {%- set status, res = adapter.execute(sql, auto_begin=auto_begin, fetch=fetch_result) -%}
-    {%- if name is not none -%}
-      {{ store_result(name, status=status, data=res) }}
-    {%- endif -%}
+    {%- set status, result = adapter.execute(sql, auto_begin=auto_begin, fetch=fetch_result) -%}
+  {%- endif -%}
 
+  {%- if name is not none -%}
+    {{ store_result(name, status=status, agate_table=result) }}
   {%- endif -%}
```

Comment:
why move this out of the execute block?

Reply:
since dbt executes macros during parsing, I figured returning None here instead of an agate table will probably break some macros. Folks are going to need to do something like

```
{%- set results = load_result('some_statement') -%}

{%- if not results %}
  {{ return(None) }}
{%- endif -%}

{{ results.some_function(...) }}
...
```

This guard around the return value from `load_result` is going to be critical if people are calling functions on a value that's supposed to be an agate table object